### PR TITLE
Simulation

### PR DESCRIPTION
**Pull request by agarny**
# Description

This PR adds a new MIME type so that we can retrieve a JSON file that decribes the UI to be generated for a given simulation dataset.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

No tests at this stage since we first need to update our two simulation datasets.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import pytest
 from app import app
 from app.config import Config
+import json
```
Comment (hsorby) [inline tests/test_api.py:4]: Why import json here? It is not used.

Comment (agarny) [reply tests/test_api.py]: Oops, forgot to remove it indeed. My bad. Fixed in 532228f.

File: app/main.py [hunk 2]
```diff
@@ -924,6 +924,22 @@ def get_available_uberonids(query):
...
+    results = process_results(dataset_search(create_pennsieve_identifier_query(identifier)))
+    results_json = json.loads(results)
+
+    try:
+        item = results_json["results"][0]
+        uri = item["s3uri"]
+        path = item["abi-simulation-file"][0]["dataset"]["path"]
+        key = f"{uri}files/{path}".replace(f"s3://{Config.S3_BUCKET_NAME}/", "")
+
+        return json.loads(direct_download_url(key))
```
Comment (hsorby) [inline app/main.py:938]: We should be using the `jsonify` function here as in line 924 to properly set headers.

**Review by hsorby — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (agarny) [reply app/main.py:938]: Agreed. Consider it done.

Otherwise, are we ok with the fact that `jsonify()` is not being used throughout the API?

Comment (agarny) [reply app/main.py]: Fixed in 2ac1a75.

**Review by hsorby — APPROVED** [review]

File: app/main.py [hunk 3]
```diff
@@ -924,6 +924,22 @@ def get_available_uberonids(query):
...
+    results = process_results(dataset_search(create_pennsieve_identifier_query(identifier)))
+    results_json = json.loads(results)
+
+    try:
+        item = results_json["results"][0]
+        uri = item["s3uri"]
+        path = item["abi-simulation-file"][0]["dataset"]["path"]
+        key = f"{uri}files/{path}".replace(f"s3://{Config.S3_BUCKET_NAME}/", "")
+
+        return jsonify(json.loads(direct_download_url(key)))
```
Comment (hsorby) [inline app/main.py:938]: Whoops missed this, should be just `jsonify(direct_download_url(key))`

**Comment by hsorby** [comment]
I think this PR is targeted at the wrong repository, shouldn't it be targeted to nih-sparc/sparc-api?

File: app/main.py (hunk 3 above)
Comment (agarny) [reply app/main.py:938]: I tried that, but it broke things for me. `direct_download_url(key)` returns a JSON string, so I believe it's right to convert it to a JSON object using `json.loads()` and then convert it to a JSON response using `jsonify()`, no?

**Comment by agarny** [comment]
> I think this PR is targeted at the wrong repository, shouldn't it be targeted to nih-sparc/sparc-api?

Yes, @alan-wu just told me that on Slack. I wasn't aware of it and will do that for any future API-related PR that I have. I understand that for this PR, @alan-wu is "ok" with this.

File: app/main.py (hunk 3 above)
Comment (hsorby) [reply app/main.py:938]: Indeed

**Pull request merged**